### PR TITLE
Remove fallbacks (3)

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/CodeStyleOptions2.cs [hunk 1]
```diff
@@ -84,88 +84,88 @@ private static PerLanguageOption2<CodeStyleOption2<bool>> CreateQualifyAccessOpt
...
         CodeStyleOptionGroups.ExpressionLevelPreferences,
         "dotnet_style_prefer_compound_assignment",
-        IdeCodeStyleOptions.CommonDefaults.PreferCompoundAssignment);
+        defaultValue: CodeStyleOption2.TrueWithSuggestionEnforcement);
 
     internal static readonly PerLanguageOption2<CodeStyleOption2<bool>> PreferSimplifiedInterpolation = CreatePerLanguageOption(
         CodeStyleOptionGroups.ExpressionLevelPreferences,
         "dotnet_style_prefer_simplified_interpolation",
-        IdeCodeStyleOptions.CommonDefaults.PreferSimplifiedInterpolation);
+        defaultValue: CodeStyleOption2.TrueWithSuggestionEnforcement);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/CodeStyleOptions2.cs:168]: i'm going to trust you inlined these all carefully and the original values were preserved.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/IdeCodeStyleOptions.cs [hunk 2]
```diff
@@ -1,103 +0,0 @@
...
-    [DataMember] public CodeStyleOption2<bool> PreferCompoundAssignment { get; init; } = CodeStyleOption2.TrueWithSuggestionEnforcement;
-    [DataMember] public CodeStyleOption2<bool> PreferSimplifiedInterpolation { get; init; } = CodeStyleOption2.TrueWithSuggestionEnforcement;
-    [DataMember] public CodeStyleOption2<UnusedParametersPreference> UnusedParameters { get; init; } = s_preferAllMethodsUnusedParametersPreference;
-    [DataMember] public CodeStyleOption2<AccessibilityModifiersRequired> AccessibilityModifiersRequired { get; init; } = s_accessibilityModifiersRequiredDefault;
-    [DataMember] public CodeStyleOption2<bool> PreferReadonly { get; init; } = CodeStyleOption2.TrueWithSuggestionEnforcement;
-    [DataMember] public CodeStyleOption2<ParenthesesPreference> ArithmeticBinaryParentheses { get; init; } = s_alwaysForClarityPreference;
-    [DataMember] public CodeStyleOption2<ParenthesesPreference> OtherBinaryParentheses { get; init; } = s_alwaysForClarityPreference;
-    [DataMember] public CodeStyleOption2<ParenthesesPreference> RelationalBinaryParentheses { get; init; } = s_alwaysForClarityPreference;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/IdeCodeStyleOptions.cs:55]: same here.  i'm going to trust these inlined properly.

**Review by CyrusNajmabadi — APPROVED** [review]
looks great.  trusting that the default values stayed the same.

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/CodeStyleOptions2.cs (hunk 1 above)
Comment (tmat) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/CodeStyleOptions2.cs:168]: I tried :). Tests actually failed in one case where I made mistake.

Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/CodeStyleOptions2.cs:168]: excellent.

**Pull request merged**